ntoskrnl/mm/ARM3/session.c

```diff
@@ -477,7 +477,7 @@ MiSessionInitializeWorkingSetList(VOID)
 
     /* Fill out the two pointers */
     MmSessionSpace->Vm.VmWorkingSetList = WorkingSetList;
-    MmSessionSpace->Wsle = (PMMWSLE)WorkingSetList->UsedPageTableEntries;
+    MmSessionSpace->Wsle = (PMMWSLE)((&WorkingSetList->VadBitMapHint) + 1);
```

Comment:
Sure this is correct for x64? Shouldn't it rather be ->CommittedPageDirectoryParents?

Reply:
Never sure of anything :wink: 
In case of the session address space, I guess that HighestUserAddress & the like don't make sense, but Page Table committed state does. This is true for process VM as well, where Wsle is set to (VmWorkingSetList + 1) while CommittedPageDirectoryParents is variable length array.
I guess we need a helper function giving the WorkingSetList size.

Reply:
Sounds good. And some comments documenting how it works would be great.